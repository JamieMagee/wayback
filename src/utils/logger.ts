import { error as _error, warning as _warn } from '@actions/core';
import chalk from 'chalk';
// eslint-disable-next-line import/default
import stripAnsi from 'strip-ansi';

function write(prefix: string, ...args: unknown[]): void {
  console.log([`[${prefix}]`, ...args].join(' '));
}

function dir(obj: unknown): void {
  console.dir(obj);
}

function debug(msg: unknown, ...args: unknown[]): void {
  write(chalk.blue('DEBUG'), msg, ...args);
}

function info(msg: unknown, ...args: unknown[]): void {
  write(chalk.green(' INFO'), msg, ...args);
}

function warn(msg: unknown, ...args: unknown[]): void {
  write(chalk.magenta(' WARN'), msg, ...args);
  _warn(stripAnsi([msg, ...args].join(' ')));
}

function error(msg: unknown, ...args: unknown[]): void {
  write(chalk.red('ERROR'), msg, ...args);
  _error(stripAnsi([msg, ...args].join(' ')));
}

const log = (m: unknown, ...args: unknown[]): void => debug(m, ...args);

log.dir = (m: unknown): void => dir(m);
log.info = (m: unknown, ...args: unknown[]): void => info(m, ...args);
log.warn = (m: unknown, ...args: unknown[]): void => warn(m, ...args);
log.error = (m: unknown, ...args: unknown[]): void => error(m, ...args);

export default log;
