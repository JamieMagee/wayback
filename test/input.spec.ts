import * as _core from '@actions/core';
import Input from '../src/input';
import { getName, mocked } from './utils';

const core = mocked(_core);

describe(getName(__filename), () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('works for multiple urls', () => {
    core.getMultilineInput.mockReturnValue(['example.com', 'example.com']);
    core.getInput
      .mockReturnValueOnce('true')
      .mockReturnValueOnce('false')
      .mockReturnValueOnce('true');
    const input = new Input();

    expect(core.getMultilineInput.mock.calls).toMatchSnapshot();
    expect(core.getInput.mock.calls).toMatchSnapshot();
    expect(input).toMatchSnapshot();
  });

  it('works for a single url', () => {
    core.getMultilineInput.mockReturnValue(['example.com']);
    core.getInput
      .mockReturnValueOnce('true')
      .mockReturnValueOnce('false')
      .mockReturnValueOnce('true');
    const input = new Input();

    expect(core.getMultilineInput.mock.calls).toMatchSnapshot();
    expect(core.getInput.mock.calls).toMatchSnapshot();
    expect(input).toMatchSnapshot();
  });

  it('throws', () => {
    core.getInput
      .mockReturnValueOnce('example.com')
      .mockReturnValueOnce('notaboolean');

    expect(() => {
      new Input();
    }).toThrow();
  });
});
