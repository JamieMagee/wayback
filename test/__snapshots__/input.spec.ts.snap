// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`undefined works for a single url 1`] = `
[
  [
    "url",
    {
      "required": true,
      "trimWhitespace": true,
    },
  ],
]
`;

exports[`undefined works for a single url 2`] = `
[
  [
    "saveErrors",
    {
      "trimWhitespace": true,
    },
  ],
  [
    "saveOutlinks",
    {
      "trimWhitespace": true,
    },
  ],
  [
    "saveScreenshot",
    {
      "trimWhitespace": true,
    },
  ],
]
`;

exports[`undefined works for a single url 3`] = `
Input {
  "saveErrors": true,
  "saveOutlinks": false,
  "saveScreenshot": true,
  "url": [
    "example.com",
  ],
}
`;

exports[`undefined works for multiple urls 1`] = `
[
  [
    "url",
    {
      "required": true,
      "trimWhitespace": true,
    },
  ],
]
`;

exports[`undefined works for multiple urls 2`] = `
[
  [
    "saveErrors",
    {
      "trimWhitespace": true,
    },
  ],
  [
    "saveOutlinks",
    {
      "trimWhitespace": true,
    },
  ],
  [
    "saveScreenshot",
    {
      "trimWhitespace": true,
    },
  ],
]
`;

exports[`undefined works for multiple urls 3`] = `
Input {
  "saveErrors": true,
  "saveOutlinks": false,
  "saveScreenshot": true,
  "url": [
    "example.com",
    "example.com",
  ],
}
`;
